app/src/main/java/com/itsaky/androidide/localWebServer/WebServer.kt

```
@@ -187,51 +248,22 @@ class WebServer(
+			while (true) {
+				val clientSocket = acceptNextClient() ?: break
 
-					// CodeRabbit objects to the following line because clientSocket may print out as "null." This is intentional. --DS
-					if (debugEnabled) log.debug("clientSocket was {}.", clientSocket)
+				// The worker owns the socket from here, including closing it. Registered before the
+				// task starts, not inside it: shutdown closes what is registered to unblock workers,
+				// and a socket registered by the worker itself can be missed in that window.
+				liveSockets.add(clientSocket)
+				try {
+					pool.execute { handleConnection(clientSocket) }
+				} catch (e: RejectedExecutionException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)